pantab/src/pantab.cpp

```diff
@@ -615,36 +615,22 @@ static auto makeReadHelper(const ArrowSchemaView *schema_view,
   }
 }
 
-static auto
-arrowTypeFromHyper(const hyperapi::SqlType &sqltype) -> enum ArrowType {
-  if (sqltype == hyperapi::SqlType::smallInt()){return NANOARROW_TYPE_INT16;}
-else if (sqltype == hyperapi::SqlType::integer()) {
-  return NANOARROW_TYPE_INT32;
-}
-else if (sqltype == hyperapi::SqlType::bigInt()) {
-  return NANOARROW_TYPE_INT64;
-}
-else if (sqltype == hyperapi::SqlType::doublePrecision()) {
-  return NANOARROW_TYPE_DOUBLE;
-}
-else if (sqltype == hyperapi::SqlType::text()) {
-  return NANOARROW_TYPE_LARGE_STRING;
-}
-else if (sqltype == hyperapi::SqlType::boolean()) {
-  return NANOARROW_TYPE_BOOL;
-}
-else if (sqltype == hyperapi::SqlType::timestamp()) {
-  return NANOARROW_TYPE_TIMESTAMP;
-}
-else if (sqltype == hyperapi::SqlType::timestampTZ()) {
-  return NANOARROW_TYPE_TIMESTAMP; // todo: how to encode tz info?
-}
-else if (sqltype == hyperapi::SqlType::date()) {
-  return NANOARROW_TYPE_DATE32;
-}
-
-throw nb::type_error(
-    ("unimplemented pandas dtype for type: " + sqltype.toString()).c_str());
+static auto arrowTypeFromHyper(const hyperapi::SqlType &sqltype)
+    -> enum ArrowType {
+      switch (sqltype.getTag()){
```

Comment:
Hmm this formatting from clang-format is terribly hard to read...wonder what is going on